### PR TITLE
Dependency cleanup!

### PR DESCRIPTION
**Pull request by capnfabs**
- Remove some unused classes
- Remove some unused dependencies
- Add comments describing what some of the libraries are for.

### CODE REVIEW

File: app/build.gradle [hunk 1]
```diff
@@ -35,24 +35,17 @@ android {
...
-    provided 'com.github.frankiesardo:auto-parcel-processor:0.2' // Automatic Parcelables
     provided 'com.squareup.dagger:dagger-compiler:1.2.2' // Dependency injection
-    // provided 'com.stanfy.spoon:spoon-gradle-plugin:0.14.1' // Test multiplexing/selection
 
     // Local .jar dependencies
     compile fileTree(dir: 'libs', include: ['*.jar'])
 
     // Local project dependencies
     compile project(':third_party:odkcollect')
-    compile project(':third_party:AutoFitTextView')
```
Comment (llvasconcellos) [inline app/build.gradle:48]: This is mentioned on the README.md

**Comment by llvasconcellos** [comment]
Besides the AutoFitTextView / README.md mention everything else seems ok.
Good to go!

File: app/build.gradle (hunk 1 above)
Comment (capnfabs) [inline app/build.gradle:48]: Hey @llvasconcellos, what do you mean by "mentioned on the README.md"? I can't see it anywhere in any README.md in the project.

**Comment by G1DR4** [comment]
Good to go

File: app/build.gradle (hunk 1 above)
Comment (llvasconcellos) [inline app/build.gradle:48]: @capnfabs How dumb am I? It's the Readme for the AutoFitTextView project. ahahahahaha.
Located in:
/third_party/AutoFitTextView/README.md
Sorry!

Comment (capnfabs) [inline app/build.gradle:48]: :stuck_out_tongue:

**Pull request merged**